pkg/cpi/commonUtils.go

```diff
@@ -39,7 +39,7 @@ func (tokenParameters TokenParameters) GetBearerToken() (string, error) {
 	resp, httpErr := httpClient.SendRequest(method, tokenFinalURL, nil, header, nil)
 	defer func() {
 		if resp != nil && resp.Body != nil {
-			resp.Body.Close()
+			CloseResponseBodyIfNecessary(resp)
 		}
 	}()
```

Comment:
With this function you can now defer with:
`defer CloseResponseBodyIfNecessary(resp)`